universe/ValueRefs.h

```diff
@@ -377,6 +424,15 @@ struct FO_COMMON_API Operation final : public ValueRef<T>
 
     unsigned int GetCheckSum() const override;
 
+    std::unique_ptr<ValueRef<T>> Clone() const override {
+        std::vector<std::unique_ptr<Operation<T>>> operands(m_operands.size());
```

Comment:
if you init with a size, it creates the objects and default inits. if you then emplace back, you'll have twice as many, the first half being all `nullptr`. you probably want to default init the `vector` and call `reserve(m_operands.size())` and then `push_back` all the clones.

Reply:
Fixed.